### PR TITLE
Change the path of config.json

### PR DESCRIPTION
**Pull request by ChihweiLHBird**

### CODE REVIEW

File: sciunit/unit_test/config_tests.py [hunk 1]
```diff
@@ -12,14 +12,17 @@ class ConfigTestCase(unittest.TestCase):
     def test_json_config(self):
         from sciunit.utils import config_get
 
-        config_path = os.path.join(sciunit.__path__[0], "config.json")
+        config_path = os.path.join(str(Path.home()), ".sciunit", "config.json")
```
Comment (rgerkin) [inline sciunit/unit_test/config_tests.py:15]: This path will not exist until it is created.  First, the file itself may not exist yet, but the .sciunit directory may not exist yet either.  I use `pathlib` to make this easier.
```
from pathlib import Path
path = Path.home() / ".sciunit" / "config.json"
path.mkdir(exist_ok=True, parents=True)
...
assert path.isfile()
```

File: sciunit/utils.py [hunk 2]
```diff
@@ -674,7 +674,18 @@ def config_get_from_path(config_path: str, key: str) -> int:
             config = json.load(f)
             value = config[key]
     except FileNotFoundError:
-        raise Error("Config file not found at '%s'" % config_path)
+        sciunit_config_dir = os.path.join(str(Path.home()), ".sciunit")
```
Comment (rgerkin) [inline sciunit/utils.py:677]: Can you switch to using `pathlib` for this section as well?

Comment (rgerkin) [reply sciunit/utils.py:677]: And everywhere in this pull request.

Comment (ChihweiLHBird) [reply sciunit/utils.py:677]: Actually, `Path` is from `pathlib`, and there is a line of code, `from pathlib import Path`, at line 38 of `utils.py`.

Would you like me to use `pathlib.Path` and i`mport pathlib`?

Comment (ChihweiLHBird) [reply sciunit/utils.py:677]: Oh, sorry, I misunderstand that. You meant from `os.path` to `pathlib`, right? I am going to do that.

File: sciunit/utils.py [hunk 3]
```diff
@@ -666,7 +666,8 @@ def create_config(data: dict) -> bool:
     try:
         cofig_dir = Path.home() / ".sciunit"
         config_path = cofig_dir / 'config.json'
-        cofig_dir.mkdir(exist_ok=True, parents=True)
+        if (not cofig_dir.is_file()):
+            cofig_dir.mkdir(exist_ok=True, parents=True)
```
Comment (rgerkin) [inline sciunit/utils.py:670]: Typo: config not cofig

File: sciunit/utils.py [hunk 4]
```diff
@@ -700,15 +701,7 @@ def config_get_from_path(config_path: str, key: str) -> int:
...
-        sciunit_config_dir = Path.home() / ".sciunit"
-        
-        if not sciunit_config_dir.is_file():
-            sciunit_config_dir.mkdir(exist_ok=True, parents=True)
-
-        config_path = sciunit_config_dir / 'config.json'
-        with open(config_path, "w") as outfile:
-            config_content = {"cmap_high": 218, "cmap_low": 38}
-            outfile.write(json.dumps(config_content))
+        create_config({"cmap_high": 218, "cmap_low": 38})
```
Comment (rgerkin) [inline sciunit/utils.py:704]: Make this dictionary a global called DEFAULT_CONFIG.  create_config should then create a config with this content if it receives no arguments.

**Pull request merged**